text/processing.py

```diff
@@ -305,3 +310,82 @@ def run(self):
                     f.write(f"{line}\n")
             else:
                 raise NotImplementedError
+
+
+class SplitTextFileJob(Job):
+    def __init__(
+        self,
+        text_file: tk.Path,
+        num_lines_per_split: int,
+        num_text_file_lines: int,
```

Comment:
In the old MT recipes we would compute such things automatically in the Job, even with another job in the init so that we could determine rqmt automatically (I am also quite sure exactly this kind of Job existed there.) I am not a fan at all of providing a meta information about the input, which can be computed by a simple command.

Reply:
But then I would need to run the command in the manager since the value would need to be computed in the `__init__` and given the dataset sizes I am working with this becomes quite slow..

Reply:
offline discussion result: make argument optional and have it computed in the job if not provided

Reply:
Sorry no, this is not what I meant, we had a "line count" job for that, it would definitely not run in the manager.

Also for me it is fine to count within the run.

Reply:
I need within the `__init__` to get the number of output files. so in run is to late

Reply:
new discussion result: add CountLinesJob and execute within the `__init__`

Reply:
we had a miscommunication 😅 

`CountLinesJob` added but removed from `SplitTextFileJob`